logdna/adapter/adapter.go

```diff
@@ -102,88 +100,81 @@ func (adapter *Adapter) getTags(m *router.Message) string {
-                        Hostname:   m.Container.Config.Hostname,
-                        Labels:     m.Container.Config.Labels,
-                    },
+        if m.Data == "" || m.Container.Name == "/logdna" {
+            continue
+        }
+
+        messageStr, err := json.Marshal(Message{
+            Message:    m.Data,
+            Container:  ContainerInfo{
+                Name:   m.Container.Name,
+                ID:     m.Container.ID,
+                Config: ContainerConfig{
+                    Image:      m.Container.Config.Image,
+                    Hostname:   m.Container.Config.Hostname,
+                    Labels:     m.Container.Config.Labels,
                 },
-                Level:      adapter.getLevel(m.Source),
-                Hostname:   adapter.getHost(m.Container.Config.Hostname),
-                Tags:       adapter.getTags(m),
-            })
-
-            if err != nil {
-                adapter.Log.Println(
-                    fmt.Errorf(
-                        "JSON Marshalling Error: %s",
-                        err.Error(),
-                    ),
-                )
-            } else {
-                adapter.Queue <- Line{
-                    Line:       string(messageStr),
-                    File:       m.Container.Name,
-                    Timestamp:  time.Now().Unix(),
-                    Retried:    0,
-                }
+            },
+            Level:      adapter.getLevel(m.Source),
+            Hostname:   adapter.getHost(m.Container.Config.Hostname),
+            Tags:       adapter.getTags(m),
+        })
+
+        if err != nil {
+            adapter.Logger.Println(
+                fmt.Errorf(
+                    "JSON Marshalling Error: %s",
+                    err.Error(),
+                ),
+            )
+        } else {
+            adapter.Queue <- Line{
+                Line:       string(messageStr),
+                File:       m.Container.Name,
+                Timestamp:  time.Now().Unix(),
             }
         }
     }
 }
 
 // readQueue is a method for reading from queue:
 func (adapter *Adapter) readQueue() {
-
     buffer := make([]Line, 0)
-    timeout := time.NewTimer(adapter.Limits.FlushInterval)
-    byteSize := 0
+    bufferSize := 0
+
+    timeout := time.NewTimer(adapter.Config.FlushInterval)
 
     for {
         select {
         case msg := <-adapter.Queue:
-            if uint64(byteSize) >= adapter.Limits.MaxBufferSize {
+            if bufferSize >= int(adapter.Config.MaxBufferSize) {
                 timeout.Stop()
                 adapter.flushBuffer(buffer)
-                timeout.Reset(adapter.Limits.FlushInterval)
                 buffer = make([]Line, 0)
-                byteSize = 0
+                bufferSize = 0
             }
 
             buffer = append(buffer, msg)
-            byteSize += len(msg.Line)
+            bufferSize += int(unsafe.Sizeof(msg.Line)) * len(msg.Line)
```

Comment:
Can you explain what you are trying to do here?

It should really only be `bufferSize += len(msg.Line)`.

Reply:
😆